tools/toolchains/arm.py

```diff
@@ -56,6 +56,11 @@ def __init__(self, target, options=None, notify=None, macros=None, silent=False,
         else:
             cpu = target.core
 
+        if not TOOLCHAIN_PATHS['ARM']:
+            exe = find_executable('armcc')
+            if exe:
```

Comment:
*__init__* is **definitely** not a constructor in the same sense as a C++ constructor. Exceptions inside constructors in C++ are bad because they might leave the object in an inconsistent state. That's hardly a concern here, when the outcome is that your program terminates with an error message. But in any case, this isn't a huge deal.